app/models/pusher.rb

```diff
@@ -289,4 +286,12 @@ def validate_spec
     policy.ui = Gem::StreamUI.new(stream, stream, stream, false)
     policy.validate(false)
   end
+
+  def serialize_spec
+    spec = self.spec.dup
```

Comment:
Why do you need to dup now and not before?

Reply:
because of the cert chain. previously we used the cert chain from the spec first, and then serialized (including calling abbreviate)